pkg/controller/velerobackup/hiveobject.go

```diff
@@ -0,0 +1,100 @@
+package velerobackup
+
+import (
+	"fmt"
+
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+)
+
+type hiveObject struct {
+	meta         *metav1.ObjectMeta
+	spec         interface{}
+	object       runtime.Object
+	checksumFunc controllerutils.ChecksumOfObjectsFunc
+	logger       log.FieldLogger
+}
+
+func newHiveObject(object runtime.Object, checksumFunc controllerutils.ChecksumOfObjectsFunc, logger log.FieldLogger) (*hiveObject, error) {
+	if object == nil {
+		return nil, fmt.Errorf("Can't convert nil to a HiveObject")
+	}
+
+	switch t := object.(type) {
+	case *hivev1.ClusterDeployment:
+		return &hiveObject{
+			meta:         &t.ObjectMeta,
+			spec:         t.Spec,
+			object:       object,
+			checksumFunc: checksumFunc,
+			logger:       logger,
+		}, nil
+	case *hivev1.SyncSet:
+		return &hiveObject{
+			meta:         &t.ObjectMeta,
+			spec:         t.Spec,
+			object:       object,
+			checksumFunc: checksumFunc,
+			logger:       logger,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("Unknown Type")
+}
+
+func (h *hiveObject) hasChanged() (bool, error) {
+	hobjLogger := addHiveObjectLoggerFields(h.logger, h)
+
+	newChecksum, err := h.calculateChecksum()
```

Comment:
Calculating the checksum should not raise an error. If it does, then this could cause a block where the Valero backup is never created because of one resource for which the checksum could not be calculated. Instead, if there is an error calculating the checksum, then a default checksum should be used instead.